pkg/synth/acl.go

```diff
@@ -124,15 +127,15 @@ func resourceRelevantToACL(e ir.ResourceType) bool {
 	return e == ir.ResourceTypeSubnet || e == ir.ResourceTypeSegment || e == ir.ResourceTypeCidr
 }
 
-func resourcesContainedInCidr(s *ir.Spec, resourceIP ir.IP, resource ir.Resource) []ir.IP {
+func subnetsContainedInCidr(s *ir.Spec, cidr *ipblock.IPBlock, resource ir.Resource) []*ipblock.IPBlock {
 	if resource.Type != ir.ResourceTypeCidr {
-		return []ir.IP{resourceIP}
+		return []*ipblock.IPBlock{cidr}
 	}
 	cidrSegmentDetails := s.Defs.CidrSegments[resource.Name]
-	cidrDetails := cidrSegmentDetails.Cidrs[ir.CidrFromIP(resourceIP)]
-	retVal := make([]ir.IP, len(cidrDetails.ContainedSubnets))
+	cidrDetails := cidrSegmentDetails.Cidrs[cidr]
```

Comment:
Are these two lookups guaranteed to be successful?

Reply:
yes, it is checked in `unmarshal` function.